Classes/Controller/UriController.php

```diff
@@ -36,27 +38,41 @@
  * @subpackage dlf
  * @access public
  */
-class UriController extends \Kitodo\Dlf\Controller\AbstractController
+class UriController extends AbstractController
 {
     /**
      * The main method of the plugin
      *
-     * @return void
+     * @return ResponseInterface
      */
-    public function mainAction()
+    public function mainAction(): ResponseInterface
     {
         // Load current document.
         $this->loadDocument();
 
         if ($this->isDocMissingOrEmpty()) {
             // Quit without doing anything if required variables are not set.
-            return;
+            return $this->htmlResponse();
         }
 
         $this->setPage();
 
         $doc = $this->document->getCurrentDocument();
 
+        $this->assignUriBook($doc);
+        $this->assignUriPage($doc);
+
+        return $this->htmlResponse();
+    }
+
+    /**
+     * Assign the uri book.
+     *
+     * @param AbstractDocument|null $doc
+     * @return void
+     */
+    public function assignUriBook(?AbstractDocument $doc): void
```

Comment:
If this functions are used only inside the `mainAction()` function then they can be made `private`.